### PR TITLE
Add a `knn` method to `elasticsearch_dsl.search.Search`

### LINKED ISSUE #1679: Does elasticsearch-dsl-py support knn_search?
Great to see that this package now supports 8+ versions of Elasticsearch ❤️ 

However, when perusing the documentation, I am not yet seeing explicit support to [knn_search similar to the lower-level python client](https://elasticsearch-py.readthedocs.io/en/latest/api.html#elasticsearch.Elasticsearch.knn_search). Curious about:

1. Is it indeed the case that knn_search is not yet supported?
2. If so, is it among the features that may be added in the near future?

Thank you very much!

### PR DESCRIPTION
**Pull request by miguelgrinberg**
Fixes #1679

This PR adds the `knn` option for k-nearest neighbor vector search.

### CODE REVIEW

**Comment by pquentin** [comment]
This looks good but I did not test locally yet. However, can you please add docs? Unfortunately, the docstring is not enough, you need to edit `docs/search_dsl.rst` too. Here's the current output: https://elasticsearch-dsl--1691.org.readthedocs.build/en/1691/search_dsl.html

File: elasticsearch_dsl/search.py [hunk 1]
```diff
@@ -494,6 +500,60 @@ def script_fields(self, **kwargs):
...
 
+    def knn(
+        self,
+        field,
+        k,
+        num_candidates,
+        query_vector=None,
+        query_vector_builder=None,
+        filter=None,
+        similarity=None,
```
Comment (pquentin) [inline elasticsearch_dsl/search.py:511]: Looks like `boost` is missing. It's specified here: https://github.com/elastic/elasticsearch-specification/blob/05c39a9dc46e107e4baaf082a6b56aae1ef5901c/specification/_types/Knn.ts#L26-L43. Documented here: https://www.elastic.co/guide/en/elasticsearch/reference/current/knn-search.html#_search_multiple_knn_fields. But weirdly missing fom https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html#search-api-knn.

**Review by pquentin — COMMENTED** [review]
Looks great! I only have a small nit.

File: elasticsearch_dsl/search.py (hunk 1 above)
Comment (miguelgrinberg) [reply elasticsearch_dsl/search.py]: @pquentin I wonder why is `boost` omitted from our documentation. But in any case, I have added it now.

**Review by pquentin — APPROVED** [review]
Thanks! LGTM.

**Comment by israellias** [comment]
We need an update for this change since this [PR](https://github.com/elastic/elasticsearch/pull/98916) has arrived at ElasticSearch [v8.12.0](https://www.elastic.co/guide/en/elasticsearch/reference/current/release-notes-8.12.0.html)

I'm trying to use "knn" inside the "query" object. But I'm getting the following error:

```
elasticsearch_dsl.exceptions.UnknownDslObject: DSL class `knn` does not exist in query.
```

How could I add a new Knn class in the accepted ones? Something like

```python
# elasticsearch-dsl/query.py

class Knn(Query):
    name = "knn"

```

**Comment by miguelgrinberg** [comment]
@israellias I have added #1770 to track this request. Thanks!

**Comment by israellias** [comment]
Thanks @miguelgrinberg
Do you have any workaround? 
I'm trying 
```python
queryset = queryset.update_from_dict(
    {
        "query": {
            "knn": {
                "field": "embedding",
                "query_vector": res.data[0].embedding,
                "filter": filter_for_knn,
                "num_candidates": 30,
            }
        },
    }
)
```

But I'm noticing that update_from_dict still calls Q which results in doing the same as
```python
queryset.query("knn", field="embedding", query_vector=res.data[0].embedding, filter=filter_for_knn, num_candidates=30)
```
and that is throwing the error since the class is simply not listed

**Pull request merged**